repository_service_tuf/cli/artifact/download.py

```diff
@@ -0,0 +1,263 @@
+
+
+def update_artifact_name_with_hash_prefix(artifact_name: str) -> str:
+    artifact_file_name = artifact_name.split("/")[-1]
+    hash_prefix = find_hash_bin(artifact_file_name)
+    new_artifact_file_name = hash_prefix + "-" + artifact_file_name
+    splitted = artifact_name.rsplit("/", 1)
+    if len(splitted) > 1:
+        return artifact_name.rsplit("/", 1)[0] + "/" + new_artifact_file_name
+    else:
+        # Annotating as it's covered, but codecov doesn't
+        # recognize that properly
+        return new_artifact_file_name  # pragma: no cover
+
+
+def decode_trusted_root(root) -> str:
+    root_encoded = bytes(root, "utf-8")
+    return base64.b64decode(root_encoded).decode("utf-8")
+
+
+def build_metadata_dir(metadata_url: str) -> str:
+    """build a unique and reproducible directory name for the repository url"""
+    name = sha256(metadata_url.encode()).hexdigest()[:8]
+    return f"{Path.home()}/.local/share/rstuf/{name}"
+
+
+def init_tofu(metadata_url: str, root_url: Optional[str]) -> bool:
+    """Initialize local trusted metadata (Trust-On-First-Use) and create a
+    directory for downloads"""
+    metadata_dir = build_metadata_dir(metadata_url)
+
+    os.makedirs(metadata_dir, exist_ok=True)
+
+    if not root_url:
+        root_url = f"{metadata_url}/1.root.json"
+    try:
+        parsed_root_url = urlparse(root_url)
+        if parsed_root_url.scheme and parsed_root_url.netloc:
+            request.urlretrieve(root_url, f"{metadata_dir}/root.json")  # nosec
+        else:
+            console.print(
+                f"Failed to parse {root_url}: ",
+                f"{parsed_root_url.scheme}, {parsed_root_url.netloc}",
+            )
+    except OSError:  # pragma: no cover
+        console.print(f"Failed to download initial root from {root_url}")
+        return False  # pragma: no cover
+
+    console.print(
+        f"Trust-on-First-Use: Initialized new root in {metadata_dir}"
+    )
+    return True
+
+
+def download_artifact(
+    metadata_url: Optional[str],
+    artifacts_url: Optional[str],
+    artifact_name: str,
+    root: Optional[str],
+    directory_prefix: Optional[str],
+) -> bool:
+    if metadata_url is None:
+        console.print("Please specify metadata url")
+        return False
+    metadata_dir = build_metadata_dir(metadata_url)
+
+    if artifacts_url is None:
+        console.print("Please specify artifacts url")
+        return False
+
+    if not os.path.isfile(f"{metadata_dir}/root.json"):
+        console.print(
+            "Trusted local root not found. Using 'tofu' to "
+            "Trust-On-First-Use or copy trusted root metadata to "
+            f"{metadata_dir}/root.json"
+        )
+        ok = init_tofu(metadata_url, root)
+        if not ok:
+            console.print(
+                f"Trusted local root not found in {metadata_url} - "
+                "`tofu` was not successful"
+            )
+            return False
```

Comment:
We discussed that instead of `console.print` + `sys.ext()` we would use the traditional way to fail with an exception https://github.com/repository-service-tuf/repository-service-tuf-cli/blob/cda764fec34c2f3371f43e72c7509d941b350a49/repository_service_tuf/cli/admin/import_artifacts.py#L174
Please change all instances where you stopped execution using `console.print()` + `sys.exit()` with an exception.